pkg/reconcilermanager/controllers/reposync_controller.go

```diff
@@ -882,7 +838,23 @@ func (r *RepoSyncReconciler) populateContainerEnvs(ctx context.Context, rs *v1be
 	return result
 }
 
-func (r *RepoSyncReconciler) validateSpec(ctx context.Context, rs *v1beta1.RepoSync, reconcilerName string) error {
+func (r *RepoSyncReconciler) validateRepoSync(ctx context.Context, rs *v1beta1.RepoSync, reconcilerName string) error {
+	if rs.Namespace == configsync.ControllerNamespace {
+		return fmt.Errorf("RepoSync objects are not allowed in the %s namespace", configsync.ControllerNamespace)
+	}
+
+	if err := validation.IsDNS1123Subdomain(reconcilerName); err != nil {
+		return fmt.Errorf("invalid reconciler name %q: %s", reconcilerName, strings.Join(err, ", "))
```

Comment:
Normally Go errors need lowercase, but i vaguely recall most of our status errors start with uppercase. I don't know if this gets prefixed or not when set as the error status... Is this going to be a consistency problem?

Reply:
Right. The message is exposed in the STALLED condition. Change it back to start with uppercase.

I believe we have other places with this consistency problem, but that can be done separately.